app/src/componentLibrary/popover/popover.jsx

```diff
@@ -52,43 +56,66 @@ export const Popover = ({
 
     const setHorizontalPosition = () => {
       switch (arrowPosition) {
-        case 'right':
+        case 'right': {
           setLeft(parentLeft + parentWidth / 2 - popoverWidth + TRIANGLE_SIZE + 16);
           break;
-        case 'middle':
+        }
+        case 'middle': {
           setLeft(parentLeft + parentWidth / 2 - popoverWidth / 2);
           break;
+        }
         case 'left':
-        default:
+        default: {
           setLeft(parentLeft + parentWidth / 2 - TRIANGLE_SIZE - 16);
+        }
       }
     };
 
     const setVerticalMiddlePosition = () => {
       setTop(parentTop + parentHeight / 2 - popoverHeight / 2);
     };
 
-    if (side === 'bottom') {
-      setTop(parentTop + parentHeight + SAFE_ZONE + TRIANGLE_SIZE);
-      setHorizontalPosition();
-    } else if (side === 'top') {
-      setTop(parentTop - SAFE_ZONE - TRIANGLE_SIZE - popoverHeight);
-      setHorizontalPosition();
-    } else if (side === 'right') {
-      setVerticalMiddlePosition();
-      setLeft(parentLeft + parentWidth + SAFE_ZONE + TRIANGLE_SIZE);
-    } else if (side === 'left') {
-      setVerticalMiddlePosition();
-      setLeft(parentLeft - SAFE_ZONE - TRIANGLE_SIZE - popoverWidth);
+    switch (side) {
+      case 'bottom': {
+        setTop(parentTop + parentHeight + SAFE_ZONE + TRIANGLE_SIZE);
+        setHorizontalPosition();
+        break;
+      }
+      case 'top': {
+        setTop(parentTop - SAFE_ZONE - TRIANGLE_SIZE - popoverHeight);
+        setHorizontalPosition();
+        break;
+      }
+      case 'right': {
+        setVerticalMiddlePosition();
+        setLeft(parentLeft + parentWidth + SAFE_ZONE + TRIANGLE_SIZE);
+        break;
+      }
+      case 'left':
+      default: {
+        setVerticalMiddlePosition();
+        setLeft(parentLeft - SAFE_ZONE - TRIANGLE_SIZE - popoverWidth);
+      }
     }
   }, [parentRef, side, arrowPosition]);
 
+  const className = cx(
+    'popover',
+    `side-${side}`,
+    `position-${arrowPosition}`,
+    variant,
+    popoverClassName,
+    {
+      [`position-${arrowVerticalPosition}`]: arrowVerticalPosition,
+    },
+  );
+
   return (
     <div
-      className={cx('popover', `side-${side}`, `position-${arrowPosition}`)}
+      className={className}
       data-automation-id={dataAutomationId}
       ref={popoverRef}
-      style={{ top, left }}
+      style={{ top: topPosition || top, left }}
```

Comment:
Maybe use topPosition above (line 41 or in switch case) so as not to do unnecessary calculations?

Reply:
Done.